src/Build/BackEnd/Components/RequestBuilder/RequestBuilder.cs

```diff
@@ -1355,6 +1358,24 @@ private void ConfigureWarningsAsErrorsAndMessages()
             }
         }
 
+        private void ConfigureKnownImmutableFolders()
+        {
+            ProjectInstance project = _requestEntry?.RequestConfiguration?.Project;
+            if (project != null)
+            {
+                //example: C:\Users\USER_NAME\.nuget\;C:\Program Files\dotnet\sdk\NuGetFallbackFolder
+                FileClassifier.Shared.RegisterImmutableDirectories(project.GetPropertyValue("NuGetPackageFolders")?.Trim());
+                // example: C:\Windows\Microsoft.NET\Framework\v4.0.30319\
+                FileClassifier.Shared.RegisterImmutableDirectories(project.GetPropertyValue("MSBuildFrameworkToolsPath32")?.Trim());
+                // example:  C:\Windows\Microsoft.NET\Framework64\v4.0.30319\
+                FileClassifier.Shared.RegisterImmutableDirectories(project.GetPropertyValue("MSBuildFrameworkToolsPath64")?.Trim());
+                // example: C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2
+                FileClassifier.Shared.RegisterImmutableDirectories(project.GetPropertyValue("FrameworkPathOverride")?.Trim());
+                // example: C:\Program Files\dotnet\
+                FileClassifier.Shared.RegisterImmutableDirectories(project.GetPropertyValue("NetCoreRoot")?.Trim());
```

Comment:
Nice find. I was curious so I tracked it down. It's generated here: https://github.com/dotnet/installer/blob/38669b06cb1816de2574123e874f9c3a5e1428fb/src/redist/targets/GenerateBundledVersions.targets#L280